src/AbstractOperations/conditional_operations.jl

```diff
@@ -0,0 +1,36 @@
+import Oceananigans.Fields: condition_operand, conditional_length
+
+# For conditional reductions such as mean(condition_operand(u * v, 0, u > 0))
+
+struct ConditionalOperation{LX, LY, LZ, O, G, C, M, T} <: AbstractOperation{LX, LY, LZ, G, T} 
+    operand :: O
+    grid :: G
+    condition :: C
+    mask :: M
+ 
+     function ConditionalOperation{LX, LY, LZ}(operand::O, grid::G, condition::C, mask::M) where {LX, LY, LZ, O, G, C, M}
+         T = eltype(operand)
+         return new{LX, LY, LZ, O, G, C, M, T}(operand, grid, condition, mask)
+     end
+end
+ 
+@inline condition_operand(operand::AbstractField, condition, mask)    = condition_operand(location(operand)..., operand.grid, condition, mask)
+@inline condition_operand(LX, LY, LZ, operand, grid, condition, mask) = ConditionalOperation{LX, LY, LZ}(operand, grid, condition, mask)
+
+@inline conditional_length(c::ConditionalOperation)       = sum(c / c)
```

Comment:
Doesn't this produce `NaN` when the `mask=0`?

Reply:
yeah, I am trying different solutions for the length. I think using `set!` we can set an auxiliary field to sum where `condition == true`

Reply:
we need to sum `OneField` with a zero mask (a field that is 1 everywhere)

Reply:
it's fast too cause `OneField` is trivial, so it all boils down to evaluating the conditional everywhere (that expense can't be avoided)